dwave/plugins/torch/tensor.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2025 D-Wave
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional
+
+import torch
+
+if TYPE_CHECKING:
+    from torch import Generator
+    from torch._prims_common import DeviceLikeType
+    from torch.types import _bool, _dtype, _size
+
+__all__ = ["randspin"]
+
+
+def randspin(
+        size: _size, *, generator: Optional[Generator] = None, dtype: Optional[_dtype] = None,
+        device: Optional[DeviceLikeType] = None, requires_grad: _bool = False, **kwargs
+) -> torch.Tensor:
+    """Returns a tensor of uniformly sampled spin values (+/-1).
+
+    Args:
+        size (tuple): A tuple defining the shape of the output tensor.
+        generator (:class:`torch.Generator`, optional): a pseudorandom number generator for sampling
+        out (Tensor, optional): the output tensor.
+        dtype (`torch.dtype`, optional) - the desired data type of returned tensor. Default: if ``None``,
+            this function returns a tensor with dtype ``torch.int64``.
+        device (:class:`torch.device`, optional): the desired device of returned tensor.
+            Default: if ``None``, uses the current device for the default tensor type
+            (see :func:`torch.set_default_device`). :attr:`device` will be the CPU
+            for CPU tensor types and the current CUDA device for CUDA tensor types.
+        requires_grad (bool, optional): If autograd should record operations on the
+            returned tensor. Default: ``False``.
+
+
+    Raises:
+        ValueError: If `low` is supplied as a keyword argument.
+        ValueError: If `high` is supplied as a keyword argument.
+
+    Returns:
+        torch.Tensor: A tensor of +/-1 values.
+    """
+    if "low" in kwargs:
+        raise ValueError("Invalid keyword argument `low`.")
+    if "high" in kwargs:
+        raise ValueError("Invalid keyword argument `high`.")
+    return 2*torch.randint(0, 2, size,
+                           generator=generator,
+                           dtype=dtype,
+                           device=device,
+                           requires_grad=requires_grad,
+                           **kwargs)-1
```

Comment:
Some cleanup needed. Spaces around `-` and `*`, and perhaps separating out the `torch.randint()` call since the final `2*` and `-1` get kind-of lost in there.

Reply:
> `-1` get lost

Changed to `b = randint ... ` and `return 2 * b - 1`

Reply:
> Also not convinced that this i needed. Couldn't you just use the regular torch.randint() and pass the results through bit2spin_soft()? Other than that this only sets low/high which isn't too cumbersome to do manually.

My motivation for adding it is similar to adding a `bit2spin` function. It's very simple to write repeatedly, but it helps with readability to have a dedicated function. What are some reasons to avoid adding this?